tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -465,96 +591,226 @@ void test_sem_take_timeout_isr(void)
+	signal_count = k_sem_count_get(&high_prio_long_sem);
+	zassert_true(signal_count == 1U, "High priority and long waiting thread"
+			" executed again");
+
 	signal_count = k_sem_count_get(&high_prio_sem);
 	zassert_true(signal_count == 1U,
-		     "Higher priority thread executed again");
+		     "Higher priority thread don't get the sem");
 
 	signal_count = k_sem_count_get(&mid_prio_sem);
-	zassert_true(signal_count == 1U,
-		     "Medium priority thread didn't get executed");
+	zassert_true(signal_count == 0U,
+		     "Medium priority thread shouldn't get the sem");
 
 	signal_count = k_sem_count_get(&low_prio_sem);
 	zassert_true(signal_count == 0U,
-		     "low priority thread shouldn't have executed");
+		     "Low priority thread shouldn't get the sem");
 
-	/* enable the low priority thread to run. */
+	/* enable the mid prio thread sem_tid_2 to run */
 	k_sem_give(&multiple_thread_sem);
 	k_sleep(K_MSEC(200));
-	/* check which threads completed. */
+
+	/* check which threads completed */
+	signal_count = k_sem_count_get(&high_prio_long_sem);
+	zassert_true(signal_count == 1U, "High priority and long waiting thread"
+			" executed again");
+
 	signal_count = k_sem_count_get(&high_prio_sem);
 	zassert_true(signal_count == 1U,
-		     "Higher priority thread executed again");
+		     "High priority thread executed again");
 
 	signal_count = k_sem_count_get(&mid_prio_sem);
 	zassert_true(signal_count == 1U,
-		     "Medium priority thread executed again");
+		     "Medium priority thread don't get the sem");
+
+	signal_count = k_sem_count_get(&low_prio_sem);
+	zassert_true(signal_count == 0U,
+		     "Low priority thread didn't get the sem");
+
+	/* enable the low prio thread(thread_1) to run */
+	k_sem_give(&multiple_thread_sem);
+	k_sleep(K_MSEC(200));
+
+	/* check the thread completed */
+	signal_count = k_sem_count_get(&high_prio_long_sem);
+	zassert_true(signal_count == 1U, "High priority and long waiting thread"
+			" executed again");
+
+	signal_count = k_sem_count_get(&high_prio_sem);
+	zassert_true(signal_count == 1U, "High priority thread executed again");
+
+	signal_count = k_sem_count_get(&mid_prio_sem);
+	zassert_true(signal_count == 1U, "Mid priority thread executed again");
 
 	signal_count = k_sem_count_get(&low_prio_sem);
 	zassert_true(signal_count == 1U,
-		     "low priority thread didn't get executed");
+			"Low priority thread don't get the sem");
+}
+
+/**
+ * @brief Test the max value a semaphore can be given and taken
+ * @details
+ * - Reset an initialized semaphore's count to zero.
+ * - Give the semaphore by a thread and verify the semaphore's count whether as
+ *   expected.
+ * - Verify the max count a semaphore can reach.
+ * - Take the semaphore by a thread and verify the semaphore's count whether
+ *   as expected.
+ * - Verify the max times a semaphore can be taken.
+ * @ingroup kernel_semaphore_tests
+ * @see k_sem_count_get(), k_sem_give()
+ */
+void test_k_sem_give_take(void)
+{
```

Comment:
"give_take" doesn't seem like an informative name for this test.  Pretty much all the tests in this file are giving and taking.

It looks like you're testing the clamping of the semaphore count?  Maybe call it "test_k_sem_count_limit()" or something?